hindsight/io_modules/derived_stream/redshift.lua

```diff
@@ -0,0 +1,95 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+local M = {}
+local ipairs = ipairs
+local read_message = read_message
+local tostring = tostring
+local type = type
+
+local string    = require "string"
+local table     = require "table"
+
+setfenv(1, M) -- Remove external access to contain everything in the module
+
+VARCHAR_MAX_LENGTH = 65535
+
+function esc_ts(v)
+    if type(v) ~= "number" then return "NULL" end
+    return string.format("(TIMESTAMP 'epoch' + %g * INTERVAL '1 seconds')", v / 1e9)
+end
+
+function esc_int(v)
+    if type(v) ~= "number" or v > 2147483647 or v < -2147483647 then
+        return "NULL"
+    end
+    return tostring(v)
+end
+
+function esc_smallint(v)
+    if type(v) ~= "number"or v > 32767 or v < -32767 then
+        return "NULL"
+    end
+    return tostring(v)
+end
+
+function esc_str(con, v, max)
+    if v == nil then return "NULL" end
+    if max == nil then max = VARCHAR_MAX_LENGTH end
+    if type(v) ~= "string" then v = tostring(v) end
+    if string.len(v) > max then v = string.sub(v, 1, max) end
+
+    local escd = con:escape(v)
```

Comment:
escape() can return a string that fails to load into the DB
PostgreSQL: ERROR:  CHAR string contains invalid ASCII character: c2 a